sdk/core/src/main/java/com/klaviyo/coresdk/config/KlaviyoConfig.kt

```diff
@@ -28,10 +28,31 @@ class MissingPermission(permission: String) : Exception("You must declare $permi
  * Stores all configuration related to the Klaviyo Android SDK.
  */
 object KlaviyoConfig : Config {
-    private const val DEBOUNCE_INTERVAL: Int = 500
-    private const val NETWORK_TIMEOUT_DEFAULT: Int = 500
-    private const val NETWORK_FLUSH_INTERVAL_DEFAULT: Int = 60000
-    private const val NETWORK_FLUSH_DEPTH_DEFAULT: Int = 20
+    /**
+     * Default value: Debounce time for fluent profile setter methods
+     */
+    private const val DEBOUNCE_INTERVAL: Int = 100
+
+    /**
+     * Default value: Network request timeout duration
+     */
+    private const val NETWORK_TIMEOUT_DEFAULT: Int = 10_000
+
+    /**
+     * Default value: Interval between flushing network queue.
+     * Also used as the basis for exponential backoff when retrying requests
+     */
+    private const val NETWORK_FLUSH_INTERVAL_DEFAULT: Int = 30_000
+
+    /**
+     * Default value: How many API requests can be enqueued before flush
+     */
+    private const val NETWORK_FLUSH_DEPTH_DEFAULT: Int = 100
```

Comment:
nit: having reviewed your PR description this is the only networking constant change I think seems a little strange. It doesn't really affect anything (as you said, your network flush interval will force the queue to flush every 30s anyway). But it just seems super unlikely to me that a single client would ever be building up a queue depth of 100 within 30s. Makes the setting seem very redundant to me, so I guess I'd choose a lower number

Reply:
Yeah good point that is way too high. For background my level of though here was: "lets see what noah did... 200, hm that sounds high I'll do half that"

Thinking it through more... The goal of depth control should be to prevent keeping the radios on for too _long_ (while the interval is to prevent turning on too _frequently_). If we figure a typical request takes ~1 second, maybe 25 is a good limit.

cc @ndurell

Reply:
Very well put. I like the conclusion a lot!

Reply:
Sorry not sure I have this setting. What does this do exactly?

Reply:
I believe this is equivalent of `let MAX_QUEUE_SIZE = 200`

Reply:
Oh sorry no it isn't... `MAX_QUEUE_SIZE` is like a guard against memory overflow on iOS. You're right you don't have this setting that I can see. 
This setting triggers the queue to be flushed once it reaches a certain _size_.

Reply:
Gee maybe I need a max queue size too...

Reply:
This is the iOS code: 
```    
    mutating func enqueueRequest(request: KlaviyoAPI.KlaviyoRequest) {
        guard queue.count + 1 < MAX_QUEUE_SIZE else {
            return
        }
        queue.append(request)
    }
```
So, appears to just drop the new request completely. We're definitely getting deep into edge case territory here but I wonder if the way to go would be to persist an overflow queue straight to disk, you can put a lot more on disk than you can just hold in memory.

Reply:
I'm thinking this might be another post-GA refinement item to circle back to, same as adding some additional special error handling.